src/fileActions.jsx

```diff
@@ -603,3 +603,30 @@ export const updateFile = (file, currentPath) => {
                     return file;
             }, exc => console.error(`Adding file ${file} failed: ${exc.toString()}`));
 };
+
+export const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
```

Comment:
I'm not super happy about greedily using superuser perms here. It makes it all too easy to download /etc/shadow or files from a different user. That's a matter of opinion though, and not a veto.

Reply:
Right, we can drop it because we have checks in place for the file permissions?

Reply:
That's the question -- coming from a classical desktop file manager this is weird, and I'd find it nice to at least point out *somehow* (icon, warning, separate "Download as administrator" etc.). But let's not block on this.